Prevent daemonizing, and ensure the child process dies on linux.

We do not care for this patch on OS X, since we always have a mDNSResponder
running there.
--- mDNSPosix/PosixDaemon.c	2011-12-02 00:39:45.000000000 +0000
+++ mDNSPosix/PosixDaemon-patched.c	2017-06-20 19:21:19.784000000 +0000
@@ -37,6 +37,9 @@
 #include <fcntl.h>
 #include <pwd.h>
 #include <sys/types.h>
+#ifdef HAVE_LINUX
+#include <linux/prctl.h>
+#endif

 #if __APPLE__
 #undef daemon
@@ -174,6 +177,10 @@

     ParseCmdLinArgs(argc, argv);

+#ifdef HAVE_LINUX
+    prctl(PR_SET_PDEATHSIG, SIGTERM);
+#endif
+
     LogMsg("%s starting", mDNSResponderVersionString);

     err = mDNS_Init(&mDNSStorage, &PlatformStorage, gRRCache, RR_CACHE_SIZE, mDNS_Init_AdvertiseLocalAddresses,
